neqo-qpack/src/decoder.rs

```diff
@@ -479,15 +478,19 @@ impl QPackDecoder {
         if cap > u64::from(self.max_table_size) {
             return Err(Error::EncoderStreamError);
         }
-        self.table.set_capacity(cap);
+        if self.table.set_capacity(cap).is_err() {
+            return Err(Error::EncoderStreamError);
+        }
```

Comment:
Suggest `map_err()?` here, if this is just converting one err type/value to another err type/value.